com.unity.ml-agents/Runtime/Sensors/BufferSensorComponent.cs

```diff
@@ -7,7 +7,7 @@ namespace Unity.MLAgents.Sensors
     /// A component for BufferSensor.
     /// </summary>
     [AddComponentMenu("ML Agents/Buffer Sensor", (int)MenuGroup.Sensors)]
-    internal class BufferSensorComponent : SensorComponent
+    public class BufferSensorComponent : SensorComponent
```

Comment:
Docstrings and/or tooltips on the public members.

Reply:
Do we need checks (either here or in `BufferSensor`) that the argument to `AppendObservation` has length `ObservableSize` ?

Reply:
I will add a check

Reply:
Add an accessor for `m_Sensor`? There's no way to directly access it (only indirectly by AppendObservation)

Reply:
Isn't createSensor enough? I do not think our SensorComponents usually give access to the underlying sensor.

Reply:
RayPerceptionSensorComponentBase does. For others, I'm not sure there's ever any way to interact with the underlying sensor; here there is (although they could also do it through the BufferSensorComponent).

I think if you don't add this, you should clean up some parts of the docs; I'll comment there.